sw/host/opentitantool/src/command/certificate.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use clap::{Args, Subcommand, ValueEnum};
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::fs;
+use std::path::PathBuf;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use ot_certs::{template, x509};
+
+/// Commands for interacting with certificates.
+#[derive(Debug, Subcommand, CommandDispatch)]
+pub enum CertificateCommand {
+    /// Generate a certificate template.
+    GenTemplate(GenTplCommand),
+}
+
+/// Generate a certificate template.
+#[derive(Clone, Debug, ValueEnum)]
+pub enum CertFormat {
+    X509,
+}
+
+/// Generate a certificate template.
+#[derive(Debug, Args)]
+pub struct GenTplCommand {
+    /// Filename of the template.
+    template: PathBuf,
+    /// Certificate format
+    #[arg(long, value_enum, default_value_t = CertFormat::X509)]
+    cert_format: CertFormat,
+    /// Output directory path.
+    output_dir: PathBuf,
+}
+
+impl CommandDispatch for GenTplCommand {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        _transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        let template_content = fs::read_to_string(&self.template).with_context(|| {
+            format!(
+                "Could not load the template file {}",
+                self.template.display()
+            )
+        })?;
+        let template =
+            template::Template::from_hjson_str(&template_content).with_context(|| {
+                format!("Failed to parse template file {}", self.template.display())
+            })?;
+        log::info!("template: {:#?}", template);
+        // NOTE for now do not clear the fields as this prevents openssl from parsing
+        // the certificate which is useful for debugging.
+        let x509 = x509::generate_certificate(&template, false)?;
+        log::info!("variables: {:#?}", x509.variables);
```

Comment:
Can you return a boxed struct here (like other commands: https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitantool/src/command/lc.rs;drc=405fcdd37826ac2872088b9716d1e336933cfe87;l=200) so that these are printed to the console in JSON form (like all the other ottool commands) instead of just logging these?

Reply:
Don't let this block the PR, but out of interest what is the use case for OTT's structured outputs?